src/OrleansProviders/Streams/Memory/MemoryAdapterFactory.cs

```diff
@@ -79,6 +72,7 @@ public void Init(IProviderConfiguration providerConfig, string name, Logger log,
 
             // 10 meg buffer pool.  10 1 meg blocks
             bufferPool = new FixedSizeObjectPool<FixedSizeBuffer>(adapterConfig.CacheSizeMb, () => new FixedSizeBuffer(1 << 20));
+            this.serializer = ActivatorUtilities.CreateInstance<TSerializer>(this.serviceProvider);
```

Comment:
hmmm, this is creating the serializer differently than in MemoryBatchContainer...